gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/AlertServiceImpl.java

```diff
@@ -410,102 +403,48 @@ public AlertEventEntity apply(AlertEvent alertEventRepo) {
-                                try {
-                                    boolean create = alertTriggerRepository
-                                        .findByReference(AlertReferenceType.API.name(), apiId)
-                                        .stream()
-                                        .noneMatch(alertTrigger -> alertId.equals(alertTrigger.getParentId()));
-
-                                    if (create) {
-                                        AlertTrigger trigger = convert(alert);
-                                        AlertTriggerEntity triggerEntity = convert(trigger);
-                                        triggerEntity.setId(UUID.toString(UUID.random()));
-                                        triggerEntity.setReferenceType(AlertReferenceType.API);
-                                        triggerEntity.setReferenceId(apiId);
-                                        triggerEntity.setTemplate(false);
-                                        triggerEntity.setEnabled(true);
-                                        triggerEntity.setEventRules(null);
-                                        triggerEntity.setParentId(alertId);
-                                        triggerEntity.setCreatedAt(new Date());
-                                        triggerEntity.setUpdatedAt(trigger.getCreatedAt());
-
-                                        create0(convert(triggerEntity));
-                                    }
-                                } catch (TechnicalException te) {
-                                    LOGGER.error("Unable to create default alert for API {}", apiId, te);
+                        apiId -> {
+                            try {
+                                boolean create = alertTriggerRepository
+                                    .findByReferenceAndReferenceId(API.name(), apiId)
+                                    .stream()
+                                    .noneMatch(alertTrigger -> alertId.equals(alertTrigger.getParentId()));
+
+                                if (create) {
+                                    AlertTrigger trigger = alertTriggerConverter.toAlertTrigger(alert);
+                                    AlertTriggerEntity triggerEntity = alertTriggerConverter.toAlertTriggerEntity(trigger);
+                                    triggerEntity.setId(UUID.toString(UUID.random()));
+                                    triggerEntity.setReferenceType(API);
+                                    triggerEntity.setReferenceId(apiId);
+                                    triggerEntity.setTemplate(false);
+                                    triggerEntity.setEnabled(true);
+                                    triggerEntity.setEventRules(null);
+                                    triggerEntity.setParentId(alertId);
+                                    triggerEntity.setCreatedAt(new Date());
+                                    triggerEntity.setUpdatedAt(trigger.getCreatedAt());
```

Comment:
Not update by you, but it's strange no ?

Reply:
Yes.
The updatedAt takes the creation date of the template (so, update date is before creation date).
I Can't explain why.